RELEASE/scripts/autoscend/quests/level_13.ash

```diff
@@ -201,8 +201,22 @@ boolean LX_getStarKey()
 		return false;
 	}
 	
+	if (can_interact())
+	{	
+		pullXWhenHaveY($item[Star Chart], 1, 0);
+		if (item_amount($item[Star]) < 8)
+		{
+			pullXWhenHaveY($item[Star], (8 - item_amount($item[Star])), item_amount($item[Star]));
+		}
+		if (item_amount($item[Star]) < 7)
+		{
+			pullXWhenHaveY($item[line], (7 - item_amount($item[line])), item_amount($item[line]));
+		}
+	}
```

Comment:
If you can_interact() you are not pulling, you are mallbuying. which should use the function
`boolean buyUpTo(int num, item it, int maxprice)`

also should avoid pullXWhenHaveY as it is obsolete since nerfday. we can only pull 1 of each unique item which means _all_ code that uses pullXWhenHaveY needs to be refactored.

Reply:
```suggestion
	if (can_interact())
	{	
		buyUpTo(1, $item[Star Chart], 1000);
		buyUpTo(8, $item[Star], 1000);
		buyUpTo(7, $item[line], 1000);
	}
```

Reply:
> I thought pulling only 1 out of ronin was fixed or about to be fixed? will do for LX_getStarKey

I am not sure what you mean by this. Nerf day made it so that in ronin, you can only pull 1 copy of an item per day (or it is per run? I need to double check).

the moment we break ronin we empty hangk into our inventory to give us access to all our items.
so out of ronin all our pull functions automatically switch over to buy.

since you required can_interact (either in casual, postronin, or aftercore) that means no matter what there are no pulls involved. just mallbuying.